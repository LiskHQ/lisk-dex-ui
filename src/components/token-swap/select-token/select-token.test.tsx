import * as React from "react";
import "@testing-library/jest-dom/extend-expect";
import SelectToken from "./select-token";
import { fireEvent, render } from "@testing-library/react";

const renderComponent = () => {
    return render(
      <SelectToken />
    );
  };

  describe("SearchBox component", () => {
    it("checks if the component matches the snapshot", () => {
      const { container } = renderComponent();
      expect(container).toMatchSnapshot();
    });  

    it("checks if  search bar is working correctly on entring textr", () => {

        const wrapper = renderComponent();
        const searchInput = wrapper.getByTestId("search-input");
        fireEvent.change(searchInput, { target: { value: "BNB" } });
        const tokenButton = wrapper.getByText("BNB");
        expect(tokenButton).toContainHTML("BNB");        
      });


    it("checks if a correct token is selected on click", () => {
        const handleTokenName = jest.fn();
        const handleSelectOpen = jest.fn();
        const handlePopularPairing = jest.fn();
        const wrapper = render(
          <SelectToken handleTokenName={handleTokenName} handleSelectOpen={handleSelectOpen} handlePopularPairing={handlePopularPairing}/>
        ); 
        const tokenButton = wrapper.getByTestId("token-name-BNB");
        fireEvent.click(tokenButton);
        expect(handleTokenName).toHaveBeenCalled();
      });


    it("checks if  correct tokens are selected on popular pairing click", () => {
        const handleTokenName = jest.fn();
        const handleSelectOpen = jest.fn();
        const handlePopularPairing = jest.fn();
        const wrapper = render(
          <SelectToken handleTokenName={handleTokenName} handleSelectOpen={handleSelectOpen} handlePopularPairing={handlePopularPairing}/>
        ); 
        const popularPairing = wrapper.getByTestId("popular-pairing-LSK,BNB");
        fireEvent.click(popularPairing);
        expect(handlePopularPairing).toHaveBeenCalled();
      });

  }); 