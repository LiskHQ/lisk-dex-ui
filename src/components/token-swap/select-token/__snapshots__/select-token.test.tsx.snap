// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox component checks if the component matches the snapshot 1`] = `
.emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

.emotion-2 {
  padding: 16px;
}

.emotion-2:last-child {
  padding-bottom: 24px;
}

.emotion-3 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-4.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-4.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-4.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-4.Mui-disabled {
  opacity: 0.38;
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #1976d2;
}

.emotion-5::-moz-focus-inner {
  border-style: none;
}

.emotion-5.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-5 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-5:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(25, 118, 210, 0.04);
}

@media (hover: none) {
  .emotion-5:hover {
    background-color: transparent;
  }
}

.emotion-5.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-6 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root makeStyles-tokenList-2 emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 makeStyles-selectToken-1 emotion-1"
    >
      Select Token
    </p>
    <div
      class="MuiCardContent-root emotion-2"
    >
      <div
        class="makeStyles-searchDiv-7"
      >
        <svg
          class="makeStyles-searchBarIcon-6"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
        <input
          class="makeStyles-searchBar-5"
          data-testid="search-input"
          placeholder="Search name or paste address"
          type="text"
        />
      </div>
      <section>
        <p
          class="makeStyles-popularPairing-8"
        >
          Popular Pairing
        </p>
        <div
          class="makeStyles-popularPairingArray-9"
        >
          <button
            class="makeStyles-popularPairingArrayItem-11"
            data-testid="popular-pairing-BTC,ETH"
          >
            BTC
            <svg
              class="makeStyles-popularPairingIcon-10"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8-8-8z"
              />
            </svg>
            ETH
          </button>
          <button
            class="makeStyles-popularPairingArrayItem-11"
            data-testid="popular-pairing-LSK,BNB"
          >
            LSK
            <svg
              class="makeStyles-popularPairingIcon-10"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8-8-8z"
              />
            </svg>
            BNB
          </button>
          <button
            class="makeStyles-popularPairingArrayItem-11"
            data-testid="popular-pairing-LUN,ETH"
          >
            LUN
            <svg
              class="makeStyles-popularPairingIcon-10"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8-8-8z"
              />
            </svg>
            ETH
          </button>
          <button
            class="makeStyles-popularPairingArrayItem-11"
            data-testid="popular-pairing-ETH,BNB"
          >
            ETH
            <svg
              class="makeStyles-popularPairingIcon-10"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8-8-8z"
              />
            </svg>
            BNB
          </button>
          <button
            class="makeStyles-popularPairingArrayItem-11"
            data-testid="popular-pairing-LSK,ETH"
          >
            LSK
            <svg
              class="makeStyles-popularPairingIcon-10"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8-8-8z"
              />
            </svg>
            ETH
          </button>
          <button
            class="makeStyles-popularPairingArrayItem-11"
            data-testid="popular-pairing-BTC,LSK"
          >
            BTC
            <svg
              class="makeStyles-popularPairingIcon-10"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8-8-8z"
              />
            </svg>
            LSK
          </button>
        </div>
        <div>
          <ul
            class="MuiList-root MuiList-padding makeStyles-cryptoListArray-3 emotion-3"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-4"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-cryptoListArrayItem-4 emotion-5"
                data-testid="token-name-Bitcoin"
                tabindex="0"
                type="button"
              >
                Bitcoin
                <span
                  class="MuiTouchRipple-root emotion-6"
                />
              </button>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-4"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-cryptoListArrayItem-4 emotion-5"
                data-testid="token-name-Ethereum"
                tabindex="0"
                type="button"
              >
                Ethereum
                <span
                  class="MuiTouchRipple-root emotion-6"
                />
              </button>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-4"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-cryptoListArrayItem-4 emotion-5"
                data-testid="token-name-Tether"
                tabindex="0"
                type="button"
              >
                Tether
                <span
                  class="MuiTouchRipple-root emotion-6"
                />
              </button>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-4"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-cryptoListArrayItem-4 emotion-5"
                data-testid="token-name-BNB"
                tabindex="0"
                type="button"
              >
                BNB
                <span
                  class="MuiTouchRipple-root emotion-6"
                />
              </button>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-4"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-cryptoListArrayItem-4 emotion-5"
                data-testid="token-name-Lisk"
                tabindex="0"
                type="button"
              >
                Lisk
                <span
                  class="MuiTouchRipple-root emotion-6"
                />
              </button>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-4"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-cryptoListArrayItem-4 emotion-5"
                data-testid="token-name-Dex"
                tabindex="0"
                type="button"
              >
                Dex
                <span
                  class="MuiTouchRipple-root emotion-6"
                />
              </button>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-4"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-cryptoListArrayItem-4 emotion-5"
                data-testid="token-name-Luna"
                tabindex="0"
                type="button"
              >
                Luna
                <span
                  class="MuiTouchRipple-root emotion-6"
                />
              </button>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-4"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-cryptoListArrayItem-4 emotion-5"
                data-testid="token-name-Terra"
                tabindex="0"
                type="button"
              >
                Terra
                <span
                  class="MuiTouchRipple-root emotion-6"
                />
              </button>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-4"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-cryptoListArrayItem-4 emotion-5"
                data-testid="token-name-Sol"
                tabindex="0"
                type="button"
              >
                Sol
                <span
                  class="MuiTouchRipple-root emotion-6"
                />
              </button>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-4"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-cryptoListArrayItem-4 emotion-5"
                data-testid="token-name-USD"
                tabindex="0"
                type="button"
              >
                USD
                <span
                  class="MuiTouchRipple-root emotion-6"
                />
              </button>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-4"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-cryptoListArrayItem-4 emotion-5"
                data-testid="token-name-XRP"
                tabindex="0"
                type="button"
              >
                XRP
                <span
                  class="MuiTouchRipple-root emotion-6"
                />
              </button>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-4"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-cryptoListArrayItem-4 emotion-5"
                data-testid="token-name-PLOK"
                tabindex="0"
                type="button"
              >
                PLOK
                <span
                  class="MuiTouchRipple-root emotion-6"
                />
              </button>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-4"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-cryptoListArrayItem-4 emotion-5"
                data-testid="token-name-Doge"
                tabindex="0"
                type="button"
              >
                Doge
                <span
                  class="MuiTouchRipple-root emotion-6"
                />
              </button>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-4"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-cryptoListArrayItem-4 emotion-5"
                data-testid="token-name-Ava"
                tabindex="0"
                type="button"
              >
                Ava
                <span
                  class="MuiTouchRipple-root emotion-6"
                />
              </button>
            </li>
          </ul>
        </div>
      </section>
    </div>
  </div>
</div>
`;
